presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -327,20 +331,23 @@ public PartitionStatistics getTableStatistics(String databaseName, String tableN
             Set<String> partitionNames,
             List<String> columnNames,
             Map<String, OptionalLong> partitionRowCounts)
+    {
+        return getMetastorePartitionColumnStatistics(databaseName, tableName, partitionNames, columnNames).entrySet().stream()
+                .filter(entry -> !entry.getValue().isEmpty())
+                .collect(toImmutableMap(
+                        Map.Entry::getKey,
+                        entry -> groupStatisticsByColumn(entry.getValue(), partitionRowCounts.getOrDefault(entry.getKey(), OptionalLong.empty()))));
```

Comment:
Assuming a canonical metastore representation of statistics exists:

Add conversion between `Map<String, Map<String, HiveColumnStatistics>>` and `Map<String, List<ColumnStatisticsObj>>` as methods `to/fromMetastoreApiColumnStatistics` in `ThriftMetastoreUtil`

Reply:
https://github.com/prestodb/presto/pull/11237#discussion_r211275141